pkg/events/queue/queue_mem_list.go

```diff
@@ -0,0 +1,109 @@
+package queue
+
+import (
+	"container/list"
+	"github.com/aquasecurity/tracee/pkg/external"
+	"sync"
+)
+
+type EventQueueMem struct {
+	amountOfEvents     int // number of cached events
+	mutex              sync.Mutex
```

Comment:
why not using `sync.RWMutex`?

Reply:
Enqueue needs list write access, dequeue also needs list write access.. Len() from `available` is read-only, so I guess we could use RO mutex there.

Reply:
Explained at: https://github.com/aquasecurity/tracee/pull/1488#discussion_r813920308